File: sdk/python/src/amane_mailer/scripts/send_local.py
"""Local Mailer smoke script for manual integration checks."""

from __future__ import annotations

import os
import sys

from amane_mailer import MailRequestBuilder, MailerClient


def main() -> int:
    api_key = os.environ.get("MAILER_API_KEY")
    if not api_key:
        print("MAILER_API_KEY must contain a managed API key.", file=sys.stderr)
        return 2

    client = MailerClient(
        base_url=os.environ.get("MAILER_BASE_URL", "http://127.0.0.1:5280"),
        bearer_token=api_key,
    )

    response = client.send_mail(
        MailRequestBuilder()
        .generate_mail_request_id()
        .purpose("FormResponseNotification")
        .to(email=os.environ.get("MAILER_RECIPIENT_EMAIL", "admin@example.com"))
        .subject("SDK smoke test")
        .text_body("Sent from amane-mailer Python SDK.")
        .build()
    )

    print(f"HTTP 202 - status: {response.status}")
    print(f"mail_request_id: {response.mail_request_id}")
    return 0


if __name__ == "__main__":
    sys.exit(main())

File: sdk/typescript/scripts/send-local.mjs
#!/usr/bin/env node
import { MailerClient, MailRequestBuilder } from '../src/index.mjs';

if (!process.env.MAILER_API_KEY) {
  throw new Error('MAILER_API_KEY must contain a managed API key.');
}

const client = new MailerClient({
  baseUrl: process.env.MAILER_BASE_URL ?? 'http://127.0.0.1:5280',
  bearerToken: process.env.MAILER_API_KEY,
});

const response = await client.sendMail(
  MailRequestBuilder.create()
    .generateMailRequestId()
    .purpose('FormResponseNotification')
    .to({ email: process.env.MAILER_RECIPIENT_EMAIL ?? 'admin@example.com' })
    .subject('SDK smoke test')
    .textBody('Sent from @amane/mailer SDK.')
    .build(),
);

console.log(`HTTP 202 - status: ${response.status}`);
console.log(`mail_request_id: ${response.mailRequestId}`);
